### DISCUSSION TITLE
Optimising Initial Load Time

### DISCUSSION THREAD

**Discussion by ShobhanSrivastava** [kinngh/shopify-node-mongodb-next-app#54] [Ideas]
Hey @kinngh, you've been a great support lately and this template has helped me a lot. 

This time I've got a suggestion in order to optimise for the initial load time. In my app, I have been getting very high initial loading time (about 5-6 seconds). I can understand it could be due to multiple factors. 

While understanding the data flow in the auth process, I saw that in the `isInitialLoad` middleware, we are registering the webhooks for the store every time where we await the `webhookRegisterResponse` which takes some time. 

In my opinion, we can move the webhook registration part to the `freshInstall` function as the webhook registration is a one time thing and we can omit the registration every time the page is loaded.

Let me know if I am missing something here as I am relatively new to Shopify App Development domain. But if you feel that this change will optimise the initial load time, I'd love to contribute to this template by making this change and creating a PR. 

Thanks
Shobhan Srivastava

**Comment by kinngh** [comment]
1. Like I stated in your X message, **do not message me saying `"I opened a new issue/discussion"`.** 
2. While understandable on a shorter term, how are you managing these scenarios:
  - Your app url / webhook urls changed
  - Your webhook gets removed because of a down time
  - You remove a webhook from your code but it's still registered 
  - You add a new webhook because your app has now evolved
 3. I do not take PRs on the repo. This is less of a collaborative open source project and more of a "This is my boilerplate that I'm giving you under an MIT license that you're free to build with".

Also,  yes, a good way to deal with 2.1,2.3 and 2.4 is to loop a GQL query and update webhooks, 2.2 where your webhooks are removed from Shopify's end will lead to a potential delisting from the app store and loosing ranks. Even having devs write a GQL query to bulk update stores' webhook endpoints is apparently a huge pain.
 
I hate the sub 13 second installation time too, but the boilerplate serves as a starting point, not a defacto `just use this and everything is taken care of`, feeding things on a silver spoon, golden bullet, but it's a small price to pay for salvation.
